java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowConfigBuilder.java

```diff
@@ -30,6 +31,8 @@
   private Calendar calendar;
   private BaseAllocator allocator;
   private boolean includeMetadata;
+  private Map<Integer, JdbcFieldInfo> arraySubTypesByColumnIndex;
```

Comment:
probably not worth changing for this review, but given this would be a relatively dense map, it might be slightly more performant to have this as an JdbcFieldInfo[]

Reply:
Depends on the number of fields returned by the query.  If a query returns 10 fields and the ninth is an array, the map wouldn't be very dense at all.  I'd prefer to leave this as-is until proven otherwise?

Reply:
Fine with leaving as is.  You are right not quite dense, but still a few number of elements in general vs boxing/unboxing cost for integers on every call (like I said probably not a performance issue at all.